jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
```

Comment:
Heh, I don't really get how it is `andThen`ing something, but `flat_map` makes sense based on the words.  :-)

A use-case for `and_then` would be something that is 'temporal', like on a promise or so, but not on an `option`.